rb/lib/selenium/webdriver/common/driver.rb

```diff
@@ -248,6 +248,42 @@ def execute_async_script(script, *args)
         bridge.execute_async_script(script, *args)
       end
 
+      #
+      # virtual-authenticator
+      #
+
+      def virtual_authenticator_id
```

Comment:
Not all the drivers support this, so I think it would be better to keep these methods in a separate module (e.g. `Selenium::WebDriver::DriverExtensions::HasVirtualAuthenticator`) and then include it in the drivers supporting it. Similar to how we work with `HasBiDi` extension.

Reply:
Could we also add some `# @example` that demonstrate how to use the APIs?

Reply:
The argument is that it is part of the spec, so the methods should be implemented across the board.

My issue is that the naming gets confusing here when tacked onto Driver class directly. I'm thinking we should have a separate class for these methods:

Something like:
```
driver.virtual_authenticator.id
driver.virtual_authenticator.add
driver.virtual_authenticator.remove
driver.virtual_authenticator.credentials=
driver.virtual_authenticator.credentials
driver.virtual_authenticator.user_verified=
```

Reply:
@p0deje as discussed with @AutomatedTester, virtual authenticator is present only in chrome for now. It may happen in future that firefox and other browsers implement it. So to keep the code changes minimal in future for other browsers, I have added virtual authenticator directly under the driver. And tests are run only in chrome, not in firefox or others. If for example, firefox decided to implement it, then we only need to include it in the test suite.

`# @examples` will be added in the documentation. That will be a separate PR. Is that what you mean?

Reply:
@titusfortner I have created a separate class `VirtualAuthenticator.rb` and all the methods have been taken there as you suggested. Please review for changes. We can now directly call:

```
authenticator.add_credential
authenticator.remove_credential
```
instead of 
```
driver.virtual_authenticator.add
driver.virtual_authenticator.remove
```

This approach will help to add multiple authenticators.